Tools/clinic/clinic.py

```diff
@@ -1217,7 +1228,11 @@ def parser_body(
         parsearg: str | None
         if not parameters:
             parser_code: list[str] | None
-            if not requires_defining_class:
+            if f.kind is GETTER:
+                flags = "NULL"
```

Comment:
This looks strange to me; AFAICS, we don't use `flags` for getters (`methoddef_flags` is not included in the template). And _if_ we were to supply a zero default for flags, it would have to be an int literal (`0`, not `NULL`).

Ideally we should not have to specify `flags` here, but a lot of the following code expect it to be defined. I suggest we either set it to the empty string or the zero int literal.

```suggestion
                flags = "0"
```


```suggestion
                flags = ""  # This should end up unused
```


@AlexWaygood?

Reply:
@erlend-aasland 
FYI, I just followed the current convention.
https://github.com/python/cpython/blob/a194938f33a71e727e53490815bae874eece1460/Modules/_io/bufferedio.c#L2543-L2548

If we decide to set 0, it will be 
`{"mode", (getter)_io__Buffered_mode_get, 0, NULL},` or `{"mode", (getter)_io__Buffered_mode_get, 0, 0}`
,

Is this what you intend? (I don't have any preference about this :) )

Reply:
Ah please ignore my previous comment, now I understood what you want to say :)

Reply:
You may already know: the reason I don't use `methodef_flags` to the template is due to the structure of `getset`.
https://github.com/python/cpython/blob/a194938f33a71e727e53490815bae874eece1460/Include/descrobject.h#L11-L17

Reply:
We're not creating a `PyMethodDef` struct here, so the template field naming is now confusing. For example, the `methoddef_name` and `methoddef_cast` names are just confusing when we're generating a `PyGetSetDef` struct. I think we should fix the naming sooner than later.

Reply:
> You may already know: the reason I don't use `methodef_flags` to the template is due to the structure of `getset`.

Yeah, I know; there are no `flags` member here. Which is why setting flags to a specific value is confusing. Future readers will wonder why it was set to `"NULL"`.

Reply:
> We're not creating a PyMethodDef struct here, so the template field naming is now confusing. For example, the methoddef_name and methoddef_cast names are just confusing when we're generating a PyGetSetDef struct. I think we should fix the naming sooner than later.

We need to refactor some of the code while implementing `@setter`, Can we handle this issue on the `@setter` PR? or Do you want to handle it as this PR?

Reply:
https://github.com/python/cpython/pull/112396/commits/dde933e3e4d73ceb7ea00a876303c8908ca795c5
Remove unnecessary template :)

Reply:
I'm not sure my suggested comment is good enough :) Wording it more explicit might be better (cc. @AlexWaygood).

Reply:
I think this is fine... or, at least, I can't think of anything better :)

Looking at this function gives me a headache; this change doesn't really make my head hurt significantly more tbh :)

Reply:
Yes, the urge to refactor is growing. I suggest we take a stab at it pretty soon, Alex :)